Source/Core/Core/PowerPC/MemoryPatches.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <cstddef>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+
+struct MemoryPatch
+{
```

Comment:
What makes this struct unique for powerpc? Can it be generalized so Core/Common does not depend on Core/PowerPC?